packages/contracts/contracts/maci/Tally.sol

```diff
@@ -0,0 +1,358 @@
+  function pause() public onlyOwner {
+    _pause();
+  }
+
+  /// @notice Unpause contract calls (deposit, claim, withdraw)
+  function unpause() public onlyOwner {
+    _unpause();
+  }
+
+  /// @inheritdoc IPayoutStrategy
+  function deposit(uint256 amount) public isInitialized whenNotPaused beforeVotingDeadline {
+    totalAmount += amount;
+
+    token.safeTransferFrom(msg.sender, address(this), amount);
+  }
+
+  /// @inheritdoc IPayoutStrategy
+  function withdrawExtra(
+    address[] calldata receivers,
+    uint256[] calldata amounts
+  ) public override isInitialized onlyOwner whenNotPaused afterCooldown {
+    uint256 amountLength = amounts.length;
+    uint256 totalFunds = totalAmount;
+    uint256 sum = 0;
+
+    for (uint256 index = 0; index < amountLength; ) {
+      uint256 amount = amounts[index];
+      address receiver = receivers[index];
+      sum += amount;
+
+      if (sum > totalFunds) {
+        revert InvalidWithdrawal();
+      }
+
+      if (amount > 0) {
+        token.safeTransfer(receiver, amount);
+      }
+
+      unchecked {
+        index++;
+      }
+    }
+
+    totalAmount -= sum;
+  }
+
+  /// @inheritdoc TallyBase
+  function addTallyResults(
+    uint256[] calldata voteOptionIndices,
+    uint256[] calldata results,
+    uint256[][][] calldata tallyResultProofs,
+    uint256 tallyResultSalt,
+    uint256 spentVoiceCreditsHashes,
+    uint256 perVOSpentVoiceCreditsHashes
+  ) public override isInitialized afterVotingDeadline onlyOwner {
+    if (recipientCount == 0) {
+      recipientCount = registry.recipientCount();
+    }
+
+    if (recipientCount == totalTallyResults) {
+      revert TooManyResults();
+    }
+
+    super.addTallyResults(
+      voteOptionIndices,
+      results,
+      tallyResultProofs,
+      tallyResultSalt,
+      spentVoiceCreditsHashes,
+      perVOSpentVoiceCreditsHashes
+    );
+
+    if (recipientCount < totalTallyResults) {
+      revert TooManyResults();
+    }
+  }
+
+  /// @inheritdoc TallyBase
+  function addTallyResult(
+    uint256 voteOptionIndex,
+    uint256 tallyResult,
+    uint256[][] calldata tallyResultProof,
+    uint256 tallyResultSalt,
+    uint256 spentVoiceCreditsHash,
+    uint256 perVOSpentVoiceCreditsHash,
+    uint8 voteOptionTreeDepth
+  ) internal override {
+    super.addTallyResult(
+      voteOptionIndex,
+      tallyResult,
+      tallyResultProof,
+      tallyResultSalt,
+      spentVoiceCreditsHash,
+      perVOSpentVoiceCreditsHash,
+      voteOptionTreeDepth
+    );
+
+    totalVotesSquares += tallyResult ** 2;
+  }
+
+  /// @inheritdoc IPayoutStrategy
+  function claim(
+    IPayoutStrategy.Claim calldata params
+  ) public override isInitialized whenNotPaused afterVotingDeadline {
+    uint256 amount = getAllocatedAmount(
+      token.balanceOf(address(this)),
+      params.voiceCreditsPerOption,
+      params.totalSpent,
+      params.tallyResult
+    );
+    totalAmount -= amount;
```

Comment:
I'm not sure I understand how this is used? this seems to be the same as token.balanceOf(address(this)), so why do we need it? or am I missing something?

Reply:
Ok, I'll replace with token balance